merchantapi/src/main/java/fi/hel/verkkokauppa/configuration/api/merchant/MerchantController.java

```diff
@@ -72,6 +73,16 @@ public ResponseEntity<String> getValue(
         );
     }
 
+    @GetMapping("/merchant/getValueDto")
```

Comment:
I added a new endpoint to return the Configuration value as actual ConfigurationDto instead of only a String. I did this because it is then uniform with the "namespace/getValue", because it returns the actual ConfigurationModel. I also got some serialization error in CommonServiceConfigurationClient when I tried to leverage the existing "merchant/getValue" to get the desired config value as String. By returning ConfigurationDto instead of a String, jackson can serialize the response to an actual object.

I did not (yet) refactor the existing "/merchant/getValue", because I was not sure how much it is used across other core and experience api services. But I will look into it more on monday and if it looks like the refactor is easy, then I shall do it. Then we can have both "/merchant/getValue" and "/namespace/getValue" endpoints to return actual ConfigurationDto:s.

Reply:
This is ok, but no need to refactor that getValue endpoint for now.